databases.go

```diff
@@ -203,6 +210,19 @@ type DatabaseUser struct {
 	Role          string                     `json:"role,omitempty"`
 	Password      string                     `json:"password,omitempty"`
 	MySQLSettings *DatabaseMySQLUserSettings `json:"mysql_settings,omitempty"`
+	KafkaSettings *DatabaseKafkaUserSettings `json:"settings,omitempty"`
```

Comment:
On `settings` vs `mysql_settings`. Plan is that `settings` should eventually be the only object we send. We're keeping `mysql_settings` around for compatibility. 

At the moment, `settings` is only kafka settings, but in the future we'll rework it so that options unique to another engine will get pulled out by a check of the engine type.